### PR TITLE
add invokeContract

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

Add `invokeContract` entry point  

## Changes

Added  `invokeContract` and doc for it.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: doc/jsonrpc-api.md [hunk 1]
```diff
@@ -325,3 +325,71 @@ Response (values omitted for brevity):
...
 ```
+
+### invokeContract
+
+Simulates a smart contract update, and returns information about the specified smart contract instance.
+
+#### Parameters
+- `blockHash` - hex encoding of a block's hash
+- `context` - a collection of parameters used to invoke a contract
+- `context.contract` - contains the index and subindex of the smart contract instance that should be invoked
```
Comment (orhoj) [inline doc/jsonrpc-api.md:336]: To remain consistent you should have:
```
context.contract.index
context.contract.subindex
```
instead to be explicit about the naming of the parameters.

File: doc/jsonrpc-api.md [hunk 2]
```diff
@@ -325,3 +325,71 @@ Response (values omitted for brevity):
...
+- `context.method` - the name of the method of the smart contract instance that should be invoked  (uint64)
```
Comment (orhoj) [inline doc/jsonrpc-api.md:337]:
```suggestion
- `context.method` - the name of the method on the smart contract instance that should be invoked
```

File: doc/jsonrpc-api.md [hunk 3]
```diff
@@ -325,3 +325,71 @@ Response (values omitted for brevity):
...
+- `context.invoker` - (Optional) contains either an account address or the index and subindex of a smart contract instance.
```
Comment (orhoj) [inline doc/jsonrpc-api.md:338]: We should have an example request for each option.

File: doc/jsonrpc-api.md [hunk 4]
```diff
@@ -325,3 +325,71 @@ Response (values omitted for brevity):
...
+- `context.parameter` - (Optional) the serialized parameter for the invoked function, defaults to no parameters.
+- `context.amount` - (Optional) the amount of microCCD's to transfer to the contract, defaults to `0`. (uint64)
+- `context.energy` - (Optional) the maximum amount of energy allowed in the execution of the contract, defaults to `10.000.000`.  (uint64)
```
Comment (orhoj) [inline doc/jsonrpc-api.md:341]:
```suggestion
- `context.energy` - (Optional) the maximum amount of energy allowed in the execution of the contract, defaults to `10.000.000`. (uint64)
```

File: src/methods.ts [hunk 5]
```diff
@@ -8,19 +8,46 @@ import {
...
+          type: 'AddressContract';
+          address: ContractAddress;
+      }
+    | {
+          type: 'AddressAccount';
+          address: string;
+      }
+    | null;
+
+interface ContractContext {
```
Comment (orhoj) [inline src/methods.ts:42]: Are these types not available from the sdk?

**Review by orhoj — CHANGES_REQUESTED** [review]

File: src/methods.ts (hunk 5 above)
Comment (shjortConcordium) [reply src/methods.ts:42]: The ContractAddress and ContractContext are not the same as the ones in the SDK. Invoker is (currently) only a local type.

Comment (orhoj) [reply src/methods.ts:42]: Okay. But please move the types to a separate file.

**Review by orhoj — APPROVED** [review]

**Pull request merged**